### PR TITLE
KAFKA-6786: Removing additional configs for StreamsBrokerDownResilienceTest

### PR DESCRIPTION
**Pull request by sh-abhi**
Updated iteration of KAFKA-6786. Currently a WIP as it is my first time contributing; want to make sure that we implemented this correctly.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by sh-abhi** [comment]
@mjsax @guozhangwang The checkstyle error should now be fixed, however I do have a concern on that failing test. It appears that the `testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup` in `ConsumerBounceTest` fails, however I only saw it fail once on my branch when I tried to run it again after seeing the Jenkins build fail. Afterwards, it always runs successfully in isolation, and as part of the larger suite.

Here is all I see as an error in my console when running locally:
```
[2019-02-03 14:54:46,773] ERROR [Consumer clientId=consumer-21, groupId=group2] Attempt to join group failed due to fatal error: Consumer group The consumer group has reached its max size. already has the configured maximum number of members. (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:552)
```

If it helps to replicate the issue I'm seeing, I'm using IntelliJ 2018.3.4, and I did try to invalidate the cache and restart, along with clearing out any output directories. 

Any suggestions? Tried using the debugger for clues, but I'm able to step through the test without any failures.

**Comment by sh-abhi** [comment]
Friendly follow up on the above @mjsax @guozhangwang

**Comment by guozhangwang** [comment]
@mjsax how about we split some reviews work here, are you able to review this PR given your workload?

**Comment by guozhangwang** [comment]
also cc @bbejeck who may also able to help reviewing.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java]: I know this wasn't there before, but maybe for this case, we should print out the argument passed.

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (bbejeck) [inline tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: Here you switched the parent class from `BaseStreamsTest` to `StreamsTestBaseService`.  That is fine, but this test relies on methods defined in `BaseStreamsTest` that will need to get added to `StreamsBrokerDownResilience`.

1. `assert_produce_consume` which itself delegates to other methods in `BaseStreamsTest`
2. There are configs specific to this test where we don't want the default Streams settings https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/streams/base_streams_test.py#L83-L86. But we also don't want to put these in `StreamsTestBaseService` as we also don't want these configs for all Streams system tests either.

Overall it might be easier to keep the parent class for `StreamsBrokerDownResilience` to `BaseStreamsTest`, and change the parent of `BaseStreamsTest` to `StreamsTestBaseService` and override the configs in `StreamsBrokerDownResilience.prop_file` like you have (including the ones mentioned above) done already. But I'll leave it up to you to decide.

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (bbejeck) [inline tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: nit: the `StreamsBrokerDownResilience` doesn't need the `UPGRADE_FROM` or `UPGRADE_TO` variables.

**Review by bbejeck — CHANGES_REQUESTED** [review]
Thanks for the contribution @sh-abhi!  
I've left some comments regarding the refactoring.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java
Comment (sh-abhi) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java]: @bbejeck nice idea! Just to be clear, I think we're suggesting that we won't print try to print the argument passed when we're printing the error?

If I'm following correctly, we can probably print out the argument passed on line 53, i.e.

```
System.out.println("StreamsTest instance started with arguments: " + Arrays.toString(args));
```

Please let me know if that sounds appropriate?

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (sh-abhi) [reply tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: Thank you for pointing this out @bbejeck. As for the decision on my end, I'll follow your suggestion as it does appear to be reasonable, and I am not yet familiar with the entire code base.

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (sh-abhi) [reply tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: Will remove.

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java]: Well. This only triggers is there are zero argument provided. So there is nothing to be printed IMHO. Maybe use:
```
System.err.println("StreamsBrokerDownResilienceTest requires one argument (properties-file) but none provided.");
```

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java]: Build fails because of this -- don't use wildcard imports.

**Review by mjsax — COMMENTED** [review]
Thanks for the new PR @sh-abhi!

Can you address the comments?

File: streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java
Comment (sh-abhi) [reply streams/src/test/java/org/apache/kafka/streams/tests/StreamsBrokerDownResilienceTest.java]: @mjsax understood; will update my IDE to avoid doing that.

**Comment by sh-abhi** [comment]
Hey @bbejeck, @mjsax, updated the PR based on the comments provided. Please take a look and let me know if anything else comes to mind?

**Comment by mjsax** [comment]
Java8 passed.

https://builds.apache.org/job/kafka-pr-jdk11-scala2.12/2547/
```
java.lang.AssertionError: Received 1, expected at least 68
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at kafka.api.ConsumerBounceTest.receiveAndCommit(ConsumerBounceTest.scala:562)
	at kafka.api.ConsumerBounceTest.$anonfun$testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup$1(ConsumerBounceTest.scala:325)
	at kafka.api.ConsumerBounceTest.$anonfun$testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup$1$adapted(ConsumerBounceTest.scala:324)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup(ConsumerBounceTest.scala:324)
```

**Comment by mjsax** [comment]
I triggered a system test run to verify the change: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2363/

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (bbejeck) [inline tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: `KafkaConfig` will need to be imported as well this line needs to go with the other imports`from kafkatest.services.kafka import KafkaConfig`

File: tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py
Comment (bbejeck) [inline tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py]: You'll need to add an import for `streams_property` with the imports above `import streams_property`

**Review by bbejeck — CHANGES_REQUESTED** [review]
Overall looks good to me, just a couple of additional minor changes are needed.

**Comment by bbejeck** [comment]
retest this please

**Comment by sh-abhi** [comment]
@bbejeck Added on the additional couple of imports.

**Comment by bbejeck** [comment]
Java 11 failed `kafka.api.UserQuotaTest.testThrottledProducerConsumer`,  Java 8 failed `kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize`

retest this please

**Review by bbejeck — COMMENTED** [review]
Thanks for the update @sh-abh changes look good, waiting for the results of this branch builder to merge https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2368/

**Review by bbejeck — APPROVED** [review]

**Comment by mjsax** [comment]
@sh-abhi System test did not pass. Can you have a look?

**Comment by sh-abhi** [comment]
@mjsax any way I can re-run the system test via Jenkins? Not sure if I would need permission to do so.

**Comment by bbejeck** [comment]
@sh-abhi I took another quick look.

I missed this before but since `propFile()` is defined in `StreamsTestBaseService` and you don't add any configs in this test you should be able to remove the overridden `propFile()` on line 45 remove the imports on line 17 and 18.
Once that is done can you ping us on this PR and we'll re-run the system-test

**Comment by bbejeck** [comment]
Hi @sh-abhi is there a status on this PR?

**Comment by sh-abhi** [comment]
@bbejeck @mjsax just updated the PR. I'll try to take a look at the Jenkins build once it completes; let me know if there are any other suggestions?

**Comment by bbejeck** [comment]
@sh-abhi thanks for the update, I've kicked off a system test run https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2504/

**Comment by sh-abhi** [comment]
@bbejeck not sure if we archive the build results somewhere; could we run it once more? Getting a 404 when I check https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2504/

**Comment by sh-abhi** [comment]
Friendly follow up on this @bbejeck

**Comment by abbccdda** [comment]
@bbejeck Could we kick off a system test build, merge this PR and close the ticket?

**Comment by bbejeck** [comment]
> @bbejeck Could we kick off a system test build, merge this PR and close the ticket?

@abbccdda ack

https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3866/

**Review by andrewchoi5 — APPROVED** [review]

**Comment by bbejeck** [comment]
@sh-abhi apologies for letting this slip.  Can you rebase this PR? Then we can get this merged.

Thanks.

**Comment by sh-abhi** [comment]
@bbejeck I'm having some difficulty doing a rebase successfully after trying some different approaches based on some references I found online. Any suggestions, or should I just open a new PR entirely?

**Comment by bbejeck** [comment]
Ok to test.

**Comment by bbejeck** [comment]
Hi @sh-abhi,

>'m having some difficulty doing a rebase successfully after trying some different approaches based on some references I found online.

This PR looks good to me, can you say what your difficulties are?

At any rate, I'll run this system test to verify the changes soon.

**Comment by sh-abhi** [comment]
@bbejeck just had some difficulty rebasing correctly with `trunk` so that we can squash our commits. However, if this isn't a concern, feel free to merge this should the system test pass.

**Comment by bbejeck** [comment]
Kicked off system test https://jenkins.confluent.io/job/system-test-kafka-branch-builder/3961/

**Comment by bbejeck** [comment]
System test failed with `Failed while trying to discover tests: Didn't find any tests for symbol tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py`.

@sh-abhi can you take a look? I'll look into this soon as well.

**Comment by mjsax** [comment]
Closing this stale PR due to inactivity. Seems the ticket still applies though?

**Pull request closed without merging**
